src/ft/mod.rs

```diff
@@ -155,6 +155,8 @@ impl Rasterize for FreeTypeRasterizer {
     }
 
     fn load_font(&mut self, desc: &FontDesc, size: Size) -> Result<FontKey, Error> {
+        // Reload font files.
+        fc::reload_config();
```

Comment:
The problem is that method takes like additional 50ms to startup time on my system. Providing additional parameter to 'not trigger this function pls' also isn't a solution. Maybe there's a way to call it on startup without overhead, but I'm not sure how (initialization is done, but it's still trying to bring things up to date).

Reply:
Ah, so the solution would be to remove this code I quoted and only have the method for reloading the cache, then call that on font (size) change?

Alternatively we could also probably try and store the age of our config and only reload if it's too old. Say for example one would load a thousand fonts, you wouldn't want to reload the config every time. So by storing a config timestamp and only updating when it's older than 'XYZ' that would probably get us most of the way there?

Reply:
The thing is that this function already does that, so in continues call of FcInitBringUptoData it'll reload font only once, if you do five calls in a raw, the problem is the first call to this function.

Reply:
Oh okay, so this wouldn't be necessary for loading a ton of fonts, but it might still help with startup as an alternative at least.

Reply:
>Alternatively we could also probably try and store the age of our config and only reload if it's too old. Say for example one would load a thousand fonts, you wouldn't want to reload the config every time. So by storing a config timestamp and only updating when it's older than 'XYZ' that would probably get us most of the way there?

I don't get it, if you download a thousand fonts and then do a config reload you expect to have all the fonts loaded(visible) to application. If you load this thousand font files one by one and update config after font update you expect a newly downloaded font be available right away. So I'm not sure what to you really want here.

The solution to make it work 'under the hood' is to allow ConfigReload only after X amount of times after startup, I think (will add a bunch of reads of 'time' though on startup).

Reply:
> I don't get it, if you download a thousand fonts and then do a config reload you expect to have all the fonts loaded(visible) to application.

My concern was not with downloading a thousand fonts online, but rather calling the `load_font` method a thousand times. As you've said that shouldn't be a problem if there's some sort of internal caching, but if there was none and you'd have to block for 50ms every time that would be a problem.

> The solution to make it work 'under the hood' is to allow ConfigReload only after X amount of times after startup, I think (will add a bunch of reads of 'time' though on startup).

That's basically accomplished by my suggestion of having an "age" associated with the actively loaded config. Since at startup the age would obviously be set to the startup time, thus the initial font load wouldn't trigger a config update.